contracts/osmosis-vault/tests/robot_tests.rs

```diff
@@ -263,6 +293,44 @@ fn deposit(different_recipient: bool, funds: Funds) {
         );
 }
 
+#[test]
+/// Tests that the first deposit works when there are already some reward assets to be compounded
+/// in the vault before the first deposit. This was a previous bug that has been fixed.
```

Comment:
how do we end up in the scenario where there are rewards prior to the first deposit? is it user1 someone enters, rewards accrue, and then user1 emergency withdraws all and so compounding is not triggered?

Reply:
Rewards can end up in the vault by someone just sending them directly to the vault without calling `ExecuteMsg::Deposit`. This is usually not done, but in this case this bug was discovered because Osmosis removed the denom creation fee, which means that the 10 OSMO that were sent to the contract upon instantiation were in there before the first deposit.